core/raftlease/store.go

```diff
@@ -196,94 +186,26 @@ func (s *Store) pinOp(operation string, key lease.Key, entity string, stop <-cha
 }
 
 func (s *Store) runOnLeader(command *Command, stop <-chan struct{}) error {
-	bytes, err := command.Marshal()
-	if err != nil {
-		return errors.Trace(err)
-	}
-	requestID := atomic.AddUint64(&s.requestID, 1)
-	responseTopic := s.config.ResponseTopic(requestID)
-
-	responseChan := make(chan ForwardResponse)
-	errChan := make(chan error)
-	unsubscribe, err := s.hub.Subscribe(
-		responseTopic,
-		func(_ string, resp ForwardResponse, err error) {
-			if err != nil {
-				errChan <- err
-			} else {
-				responseChan <- resp
-			}
-		},
-	)
-	if err != nil {
-		return errors.Annotatef(err, "running %s", command)
-	}
-	defer unsubscribe()
-
-	start := time.Now()
+	start := s.config.Clock.Now()
 	defer func() {
-		elapsed := time.Since(start)
+		elapsed := s.config.Clock.Now().Sub(start)
 		logger.Tracef("runOnLeader %v, elapsed from publish: %v", command.Operation, elapsed.Round(time.Millisecond))
 	}()
 
-	delivered, err := s.hub.Publish(s.config.RequestTopic, ForwardRequest{
-		Command:       string(bytes),
-		ResponseTopic: responseTopic,
-	})
-	if err != nil {
-		s.record(command.Operation, "error", start)
-		return errors.Annotatef(err, "publishing %s", command)
-	}
-
-	// First block until subscribers are notified.
-	// In practice, this will be the Raft forwarder running on the leader node.
-	// This is an explicit step so that we can more accurately diagnose issues
-	// in-theatre.
-	select {
-	case <-pubsub.Wait(delivered):
-	case <-s.config.Clock.After(s.config.ForwardTimeout):
-		logger.Warningf("delivery timeout waiting for %s to be processed", command)
-		s.record(command.Operation, "delivery timeout", start)
-		return lease.ErrTimeout
-	}
+	ch := make(chan struct{})
+	defer close(ch)
 
-	// Now wait for the response.
-	// The timeout starts again here, which is deliberate.
-	// It is the same timeout that is used by the Raft forwarder
-	// when `Apply` is called on the FSM.
-	select {
-	case response := <-responseChan:
-		err := RecoverError(response.Error)
-		logger.Tracef("got response, err %v", err)
-		result := "success"
-		if err != nil {
-			logger.Warningf("command %s: %v", command, err)
-			result = "failure"
+	ctx, cancel := context.WithCancel(context.TODO())
```

Comment:
We could just pass stop here, I was just playing around with a cleaner interface when we swap out of an API request.